animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+  have multiple routes. To specify a route, use
+  [MaterialPageRoute,](https://docs.flutter.io/flutter/material/MaterialPageRoute-class.html)
+  [CupertinoPageRoute,](https://docs.flutter.io/flutter/cupertino/CupertinoPageRoute-class.html)
+  or build a custom route using
+  [PageRouteBuilder.](https://docs.flutter.io/flutter/widgets/PageRouteBuilder-class.html)
+
+## Flying transitions
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* Specify a route using MaterialPageRoute, CupertinoPageRoute,
+  or PageRouteBuilder. The flying examples shown here use
+  MaterialPageRoute.
+* Change the size of the image at the end of the transition by
+  wrapping the destination's image in a SizedBox.
+* Change the location of the image by placing the destination's
+  image in a layout widget. These examples use Container.
+</div>
+
+<aside class="alert alert-info" markdown="1">
+**Show me the code!**
+
+Each of the following examples demonstrate flying an element from one
+route to another. This guide describes the first example.<br><br>
+
+[flying_transition](https://github.com/flutter/website/tree/master/_includes/code/animation/flying_transition)
+: Movement along a curved path, as described in the Material motion
+  spec. Encapsulates the Hero code in a custom PhotoHero widget.
+
+[basic_flying_transition](https://github.com/flutter/website/tree/master/_includes/code/animation/basic_flying_transition)
+: Uses the Hero object directly.
+  This more basic example, provided for your reference, isn't
+  described in this guide.
+
+</aside>
+
+Flying an element from one route to another, one of the most
+common types of shared element transitions, is easy to implement
+using Flutter's Hero widget. When using MaterialPageRoute
+to specify the new route, the element flies along a curved path,
+as described by the [Material Design motion
+spec.](https://material.io/guidelines/motion/movement.html)
+The widget tree for the destination route is created in the `builder`
+property.
+
+[Create a new Flutter example](/getting-started/) and
+update it using the files from the
+[GitHub directory.](https://github.com/flutter/website/tree/master/_includes/code/animation/flying_transition)
+
+To run the example:
+
+* Tap on the home route’s photo to fly the image to a new route
+  showing the same photo at a different location and scale.
+* Return to the previous route by tapping the image, or by using the
+  device’s back-to-the-previous-screen gesture.
+* You can slow the transition further using the `timeDilation`
+  property.
+
+### PhotoHero class
+
+The custom PhotoHero class maintains the Hero, and its size, image,
+and behavior when tapped. The PhotoHero builds the following
+widget tree:
+
+<img src="images/photohero-class.png" alt="widget tree for the PhotoHero class">
```

Comment:
I think this diagram (or some text nearby) is the only place where it's worth mentioning the InkWell and the rationale for the Material widget (we don't want the splash to be left behind).